test/tests/Pathfinding.cpp

```diff
@@ -0,0 +1,237 @@
+
+    static ::testing::AssertionResult AssertPositionIsSetUp(const char* positionKind, uint32_t expectedSurfaceStyle, const TileCoordsXYZ& location)
+    {
+        const uint32_t style = map_get_surface_element_at(location.x, location.y)->AsSurface()->GetSurfaceStyle();
+
+        if (style != expectedSurfaceStyle)
+            return ::testing::AssertionFailure()
+                << positionKind << " location " << location << " should have surface style " << expectedSurfaceStyle
+                << " but actually has style " << style
+                << ". Either the test map is not set up correctly, or you got the coordinates wrong.";
+
+        return ::testing::AssertionSuccess();
+    }
+
+    static std::shared_ptr<IContext> _context;
+};
+
+std::shared_ptr<IContext> PathfindingTestBase::_context;
+
+struct SimplePathfindingScenario
+{
+    const char* name;
+    TileCoordsXYZ start;
+    TileCoordsXYZ goal;
+    uint32_t steps;
+
+    SimplePathfindingScenario(const char* _name, const TileCoordsXYZ& _start, const TileCoordsXYZ& _goal, int _steps = 10000)
+        : name(_name)
+        , start(_start)
+        , goal(_goal)
+        , steps(_steps)
+    {
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, const SimplePathfindingScenario& scenario)
+    {
+        return os << scenario.start << " => " << scenario.goal;
+    }
+
+    static std::string ToName(const ::testing::TestParamInfo<SimplePathfindingScenario>& param_info)
+    {
+        return param_info.param.name;
+    }
+};
+
+class SimplePathfindingTest : public PathfindingTestBase, public ::testing::WithParamInterface<SimplePathfindingScenario>
+{
+};
+
+TEST_P(SimplePathfindingTest, CanFindPathFromStartToGoal)
+{
+    const SimplePathfindingScenario& scenario = GetParam();
+
+    ASSERT_PRED_FORMAT1(AssertIsStartPosition, scenario.start);
+    ASSERT_PRED_FORMAT1(AssertIsGoalPosition, scenario.goal);
+
+    TileCoordsXYZ pos = scenario.start;
+
+    const auto succeeded = FindPath(&pos, scenario.goal, scenario.steps) ? ::testing::AssertionSuccess()
+                                                                         : ::testing::AssertionFailure()
+            << "Failed to find path from " << scenario.start << " to " << scenario.goal << " in " << scenario.steps
+            << " steps; reached " << pos << " before giving up.";
+
+    EXPECT_TRUE(succeeded);
+}
+
+INSTANTIATE_TEST_CASE_P(
+    ForScenario, SimplePathfindingTest,
+    ::testing::Values(
+        SimplePathfindingScenario("StraightFlat", { 2, 19, 14 }, { 4, 19, 14 }, 24),
+        SimplePathfindingScenario("SBend", { 2, 17, 14 }, { 4, 16, 14 }, 39),
+        SimplePathfindingScenario("UBend", { 2, 14, 14 }, { 2, 12, 14 }, 88),
+        SimplePathfindingScenario("CBend", { 2, 10, 14 }, { 2, 7, 14 }, 133),
+        SimplePathfindingScenario("TwoEqualRoutes", { 6, 18, 14 }, { 10, 18, 14 }, 819),
+        SimplePathfindingScenario("TwoUnequalRoutes", { 6, 14, 14 }, { 10, 14, 14 }, 15643),
```

Comment:
15643 steps is a rather large value. Is it the first time the peep has entered the goal tile?

Reply:
Good point... I just plugged in the number it was giving me but this needs further investigation. If the peep took some action other than walking then it might explain it, but equally it could be that they walked around in a loop instead of taking the expected turn towards the goal position...

Reply:
Turns out that because the peep isn't heading to a ride, they use 'aimless' pathfinding when presented with a path junction, which means they're not moving into the requested destination cell. Might need to rework the tests so that we have actual rides as destinations instead of just coordinates...

Reply:
OK, reworked in f3776c3. As a side effect, this now means that each little path situation in the test data now has a ride which has the name of the corresponding test.